src/audio/chain_dma.c

```diff
@@ -37,56 +48,478 @@ static struct comp_dev *chain_dma_create(const struct comp_driver *drv,
 	if (!cd)
 		goto error;
 
-	cd->host_connector_node_id = cdma->primary.r.host_dma_id;
-	cd->link_connector_node_id = cdma->primary.r.link_dma_id;
 	cd->first_data_received = false;
 	cd->under_over_run_notification_sent = false;
+	cd->cs = scs ? 2 : 4;
+	cd->chain_task.state = SOF_TASK_STATE_INIT;
+
 	comp_set_drvdata(dev, cd);
 
+	ret = chain_task_init(dev, host_dma_id, link_dma_id, fifo_size);
+	if (ret)
+		goto error;
+
 	return dev;
 
 error:
 	rfree(dev);
 	return NULL;
 }
 
-int chain_dma_start(struct comp_dev *dev)
+int chain_task_start(struct comp_dev *dev)
 {
+	struct chain_dma_data *cd = comp_get_drvdata(dev);
+	k_spinlock_key_t key;
+	struct comp_driver_list *drivers = comp_drivers_get();
+	int ret;
```

Comment:
Please re-arrange the variables to reverse xmas tree (declarations longest to shortest)